lib/common/grpc-service.js

```diff
@@ -266,6 +268,25 @@ GrpcService.prototype.request = function(protoOpts, reqOpts, callback) {
     request: function(_, onResponse) {
       respError = null;
 
+      if (stream) {
+        service[protoOpts.method](reqOpts, grpcOpts)
+          .pipe(stream)
```

Comment:
I think the issue here is that grpc streams don't emit a `response` event (at least, not that I can tell). In `retry-request` the data isn't released to the user-held stream until after this event is fired.